wallet/wallet-controller/src/lib.rs

```diff
@@ -36,14 +38,27 @@ pub enum ControllerError {
 
 pub struct Controller<T> {
     rpc_client: T,
-    _wallet: DefaultWallet,
+    _wallet: Arc<Mutex<DefaultWallet>>,
```

Comment:
Why do we need `Arc<Mutex>` here? I would really not have to do this early on instead of making this technical debt.

Reply:
`Mutex` is not used now.